pageserver/src/utilization.rs

```diff
@@ -0,0 +1,40 @@
+//! An utilization metric which is used to decide on which pageserver to put next tenant.
+//!
+//! The metric is exposed via `GET /v1/utilization`. Refer and maintain it's openapi spec as the
+//! truth.
+
+use anyhow::Context;
+use std::path::Path;
+
+use pageserver_api::models::PageserverUtilization;
+
+/// u64 in json is problematic for many parsers. This is 2**53 - 1, which will be equal when parsed
+/// as u64 or f64.
+const MAX_JSON_INTEGER: u64 = 9007199254740991;
```

Comment:
This is just [common problem](https://stackoverflow.com/questions/3793838/which-is-the-first-integer-that-an-ieee-754-float-is-incapable-of-representing-e) in languages where all numbers are `f64`. It would allow to work around any json parser problems except those which you mentioned in the other thread defaulting integers to 32-bit integers. I did not see that coming :)

Reply:
I dropped the safe json integer in https://github.com/neondatabase/neon/commit/3140f6486e95f82b84455a5dc9d2551034f5e6ef in favor of using u64::MAX internally along with `format: int64`, served outside as u63::MAX or i64::MAX.